cmake/OpenCVFindLAPACK.cmake

```diff
@@ -108,12 +108,9 @@ macro(ocv_lapack_check)
 
     set(LAPACK_TRY_COMPILE_DEF "")
     if(LAPACK_IMPL STREQUAL "LAPACK/Apple" AND NOT IOS) # https://github.com/opencv/opencv/issues/24660
-      # Get macOS version
-      execute_process(COMMAND sw_vers -productVersion
-                      OUTPUT_VARIABLE MACOS_VERSION
-                      OUTPUT_STRIP_TRAILING_WHITESPACE)
-      # Enable Accelerate New LAPACK if macOS >= 13.3
-      if (MACOS_VERSION VERSION_GREATER "13.3" OR MACOS_VERSION VERSION_EQUAL "13.3")
+      # Enable Accelerate New LAPACK if target macOS >= 13.3
+      if(${CMAKE_OSX_DEPLOYMENT_TARGET} VERSION_GREATER "13.3" OR ${CMAKE_OSX_DEPLOYMENT_TARGET} VERSION_EQUAL "13.3")
+        message(STATUS "LAPACK(${LAPACK_IMPL}): Accelerate New LAPACK is enabled.")
```

Comment:
Something else has just occurred to me: this won't work on other Darwin variants (iOS, tvOS, visionOS, etc) because they have different version numbers.

For this reason, the `NOT IOS` check would be better as `IS MACOS`.

Another alternative is not to do any of this guessing and just let the user/programmer define `ACCELERATE_NEW_LAPACK` and `ACCELERATE_LAPACK_ILP64` if he wants them.

The most important part is that OpenCV's code contains various `#ifdef ACCELERATE_NEW_LAPACK` conditions to support thee newer mode (which I believe you did already).

Reply:
Friendly ping. (Our build is broken because of this...)

@fengyuentau maybe it would be good to squash your two commits?